packages/dds/migration-shim/src/sharedTreeDeltaHandler.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils";
+import { type IDeltaHandler } from "@fluidframework/datastore-definitions";
+import { type ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { type IShimDeltaHandler } from "./types";
+
+/**
+ * Handles incoming and outgoing deltas/ops for the SharedTreeShim distributed data structure.
+ * This serves as an adapter to the real DeltaHandler, filter/process migration ops
+ *
+ * TODO: Needs to be able to process v1 and v2 ops, differentiate between them, understand the various states
+ * and drop v1 ops after migration. After the MSN of the barrier op, it needs to process v2 ops without needing to
+ * check for the v2 stamp.
+ */
+export class SharedTreeShimDeltaHandler implements IShimDeltaHandler {
```

Comment:
Why do we need two different handlers?

Reply:
We technically don't need two, we could just use 1 and add a bunch of checks into it. Personally I think that's harder to think about and reason about. It also makes it easier to work on different parts of the code simultaneously so I can put out multiple PRs without them conflicting with each other.

Reply:
I don't want to create a multiple state machine just so that I can have one class. I could have multiple classes each handling a different state.